### PR TITLE
Qt fix to wxCheckListBox so that the wxEVT_CHECKLISTBOX event gets se…

### PR DESCRIPTION
**Pull request by davidatroberts**
…nt when a checkbox has been clicked

### CODE REVIEW

File: src/qt/listbox.cpp [hunk 1]
```diff
@@ -21,29 +21,49 @@ class wxQtListWidget : public wxQtEventSignalHandler< QListWidget, wxListBox >
...
 {
     wxListBox *handler = GetHandler();
     if ( handler )
-        handler->QtSendEvent(wxEVT_LISTBOX_DCLICK, index, true);
+        handler->QtSendEvent(wxEVT_LISTBOX_DCLICK, index.row(), true);
 }
 
+void wxQtListWidget::itemChanged(QListWidgetItem *item)
+{
+    if (item->flags() & Qt::ItemIsUserCheckable)
```
Comment (catalinr) [inline src/qt/listbox.cpp]:
```suggestion
    if ( item->flags() & Qt::ItemIsUserCheckable )
```

File: src/qt/listbox.cpp [hunk 2]
```diff
@@ -21,29 +21,49 @@ class wxQtListWidget : public wxQtEventSignalHandler< QListWidget, wxListBox >
...
+    {
+        wxListBox *handler = GetHandler();
+        if (handler)
```
Comment (catalinr) [inline src/qt/listbox.cpp]:
```suggestion
        if ( handler )
```

File: src/qt/listbox.cpp [hunk 3]
```diff
@@ -21,29 +21,49 @@ class wxQtListWidget : public wxQtEventSignalHandler< QListWidget, wxListBox >
...
+        {
+            QWidget *widgetHandle = handler->GetHandle();
+            QListWidget *listWidget = dynamic_cast<QListWidget*>(widgetHandle);
+
+            if (listWidget)
```
Comment (catalinr) [inline src/qt/listbox.cpp]:
```suggestion
            if ( listWidget )
```

File: src/qt/listbox.cpp [hunk 4]
```diff
@@ -21,29 +21,49 @@ class wxQtListWidget : public wxQtEventSignalHandler< QListWidget, wxListBox >
...
+            {
+                const int rowIndex = listWidget->row(item);
```
Comment (catalinr) [inline src/qt/listbox.cpp]:
```suggestion
                int rowIndex = listWidget->row(item);
```

File: src/qt/listbox.cpp [hunk 5]
```diff
@@ -21,29 +21,49 @@ class wxQtListWidget : public wxQtEventSignalHandler< QListWidget, wxListBox >
...
 
+void wxQtListWidget::itemChanged(QListWidgetItem *item)
+{
+    if (item->flags() & Qt::ItemIsUserCheckable)
+    {
+        wxListBox *handler = GetHandler();
+        if (handler)
+        {
+            QWidget *widgetHandle = handler->GetHandle();
+            QListWidget *listWidget = dynamic_cast<QListWidget*>(widgetHandle);
```
Comment (catalinr) [inline src/qt/listbox.cpp]: Is there any other QWidget that can possibly be wrapped by `wxListBox`?

Comment (davidatroberts) [reply src/qt/listbox.cpp]: There's `QListView `that could be used, although it's almost identical except in how it binds to data

Comment (catalinr) [reply src/qt/listbox.cpp]: Hold on, I'm getting really confused here with all these `handle`s and `handler`s.
We're in a `wxQtListWidget` here, which derives from `QListWidget`, which pretty much makes `listWidget` == `this`, doesn't it? So are all the `GetHandler()`, `GetHandle()`, `dynamic_cast()` and `if()`s really needed?

Comment (davidatroberts) [reply src/qt/listbox.cpp]: True, the `getHandle` and `dynamic_cast` could be removed and just call `this->row(item)` instead. I think the `GetHandler` is still needed in order to call `QtSendEvent`

**Comment by vadz** [comment]
Thanks for this fix! I've pushed it to the qt-fixes branch of #1552 and will merge it in master soon.

**Pull request closed without merging**